tests/testthat/test-age_calculate.R

```diff
@@ -0,0 +1,77 @@
+test_that("Age is calculated correctly for all units", {
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "years", date_class = "period",
+                             round_down = TRUE), 1)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "months", date_class = "period",
+                             round_down = TRUE), 12)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "weeks", date_class = "period",
+                             round_down = TRUE), 55)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "days", date_class = "period",
+                             round_down = TRUE), 385)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "hours", date_class = "period",
+                             round_down = TRUE), 9240)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "minutes", date_class = "period",
+                             round_down = TRUE), 554400)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "seconds", date_class = "period",
+                             round_down = TRUE), 33264000)
+})
+
+test_that("Age is calculated correctly for leap years", {
+  expect_equal(age_calculate(as.Date("2020-02-29"), as.Date("2022-02-28"),
+                             units = "years", date_class = "period",
+                             round_down = TRUE), 1)
+  expect_equal(age_calculate(as.Date("2020-02-29"), as.Date("2022-03-01"),
+                             units = "years", date_class = "period",
+                             round_down = TRUE), 2)
+})
+
```

Comment:
Probably prudent to have some additional tests here e.g. counts of days with a leap day in it - "2020-02-20":"2020-03-01" should be 10 days.